### DISCUSSION TITLE
SCM Menu Support

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11311] [Q&A] [answered]
Hi,
I'm trying to bring Git and SCM Commands to menubar. I can see that the Git Commands are contributed by VSCode's Git Plugin. Is there any way I could bring those commands to the Menubar ?

Thanks

**Comment by msujew** [comment, marked as answer]
Hi @maheshrajrp,

commands from plugins are handled in the same way as "native" commands in regards to menus/context-menus. You can simply add them using the [mechanism outlined in the docs](https://theia-ide.org/docs/commands_keybindings/#contributing-menu-items). You can find the command names in the `package.json` of the vscode extension. I just tried it out with the `emmet` extension:

<details>

```ts
menus.registerMenuAction(CommonMenus.FILE_CLOSE, {
  commandId: 'editor.emmet.action.wrapIndividualLinesWithAbbreviation'
});
```

![image](https://user-images.githubusercontent.com/4377073/174582951-dfd0c985-51dd-4a30-a347-f5935d9141ef.png)


</details>

**Reply by maheshrajrp** [reply]
This works !! Thanks.
